### PR TITLE
Add metrics package.

### PR DESCRIPTION
**Pull request by alyssa**

### CODE REVIEW

File: metrics/metrics.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
+// Copyright 2017, Square, Inc.
+
+// Package metrics is a wrapper of github.com/rcrowley/go-metrics.
```
Comment (daniel-nichter) [inline metrics/metrics.go]: At the risk of being pedantic :-) I'd rephrase this. This doc line is supposed to say what the pkg provides. Being a wrapper is what the pkg _is_. It's also an implementation detail. For example, our next intern will come across this and wonder what this package does (i.e. provides). (Then after, they might want to know how it does that, but that's where the code itself comes in.)

File: metrics/metrics.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+package metrics
+
+import (
+	gometrics "github.com/rcrowley/go-metrics"
+)
+
+type Metrics interface {
+	GetOrRegister(string, interface{})
```
Comment (daniel-nichter) [inline metrics/metrics.go]: Is this method overloaded? i.e. doing multiple, discrete things? Also, what's being gotten if there's no return value?

File: metrics/metrics.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+type counter struct {
+	counter gometrics.Counter
+}
+
+type Timer interface {
+	Time(func())
+	Percentile(float64) float64
+	Mean() float64
+	Count() int64
```
Comment (daniel-nichter) [inline metrics/metrics.go]: Total time?

File: metrics/metrics.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	timer gometrics.Timer
+}
+
+func NewMetrics() Metrics {
+	return &metrics{
+		registry: gometrics.NewRegistry(),
+	}
+}
+
+// Gets an existing metric or creates and registers a new one.
```
Comment (daniel-nichter) [inline metrics/metrics.go]: I usually look for method docs at the interface definition. Granted, the Go report card will complain unless both places are documented, but imho the interface def is the best place because it's, well, _the_ interface def.

File: metrics/metrics.go (hunk 1 above)
Comment (alyssa) [reply metrics/metrics.go]: I see where you're coming from. There's a difference between what the package is vs what is its purpose.

What do you think of the following?

Package metrics provides a threadsafe way to manage counters and timers.

File: metrics/metrics.go (hunk 2 above)
Comment (alyssa) [reply metrics/metrics.go]: I misread the docs, it is supposed to return something. I'll rework it. But it is supposed to do get or register in one call because that makes it threadsafe, so it's not overloaded in that sense.

File: metrics/metrics.go (hunk 3 above)
Comment (alyssa) [reply metrics/metrics.go]: I don't see what the total time would be useful for (assuming that you mean total time to be the sum of all the times we've recorded). Why do you see it useful to report?

File: metrics/metrics.go (hunk 4 above)
Comment (alyssa) [reply metrics/metrics.go]: ok, will update

File: metrics/metrics.go (hunk 3 above)
Comment (daniel-nichter) [reply metrics/metrics.go]: Sorry, I meant: "Does this return Total Time?"  i.e. at first, I don't know what a Time.Count would be. I'm playing devil's advocated. Later I saw it's count of events.

Comment (alyssa) [reply metrics/metrics.go]: Ah, I see. Yeah, its's the count of events. There's a comment on the Count() function, so that should be ok?

Comment (alyssa) [reply metrics/metrics.go]: Or actually I can move the code comment from above "Count()" method to inside the interface definition here.

**Comment by daniel-nichter** [comment]
Have been thinking and reading broadly about metrics. Going to implement this in different way.
